### PR TITLE
Fix retention boundary so 2h retention deletes blocks right at the 2h boundary

### PR DESCRIPTION
**Pull request by darshanime**
Signed-off-by: darshanime <deathbullet@gmail.com>
closes https://github.com/prometheus/prometheus/issues/9595

### CODE REVIEW

**Review by codesome — COMMENTED** [review]
Since there was no test failure, this change deserves a unit test so that the behaviour remains same in the future :) (Also, a rebase with main will mostly fix the windows test)

**Comment by beorn7** [comment]
@darshanime sorry for the blast from the past. We have discussed this during our weekly bug scrub. This change seems fine, but it needs a unit test, as @codesome said. Are you still up to finalize this?

**Comment by bboreham** [comment]
Discussed again at the bug-scrub: we want the change but also want a test.
@jesusvazquez could you take a look?

File: tsdb/db_test.go [hunk 1]
```diff
@@ -682,6 +682,32 @@ func TestDB_Snapshot(t *testing.T) {
...
 
+func TestDB_BeyondTimeRetention(t *testing.T) {
+	opts := DefaultOptions()
+	opts.RetentionDuration = 100
+	db := openTestDB(t, opts, nil)
+	defer func() {
+		require.NoError(t, db.Close())
+	}()
+
+	metas := []BlockMeta{
```
Comment (bboreham) [inline tsdb/db_test.go:693]: I would like a little explanation what is going on here - why these 4 blocks, why in this order, what are we expecting.

**Review by bboreham — APPROVED** [review]
Need to remember to call out the change in behaviour in the release notes.

File: tsdb/db_test.go (hunk 1 above)
Comment (darshanime) [reply tsdb/db_test.go:694]: added some notes

**Review by jesusvazquez — APPROVED** [review]
LGTM! Thank you @darshanime for the contribution

**Pull request merged**